executor/checksum.go

```diff
@@ -187,27 +187,47 @@ func newChecksumContext(db *model.DBInfo, table *model.TableInfo, startTs uint64
 }
 
 func (c *checksumContext) BuildRequests(ctx sessionctx.Context) ([]*kv.Request, error) {
-	reqs := make([]*kv.Request, 0, len(c.TableInfo.Indices)+1)
-	req, err := c.buildTableRequest(ctx)
-	if err != nil {
+	var partDefs []model.PartitionDefinition
+	if part := c.TableInfo.Partition; part != nil {
+		partDefs = part.Definitions
+	}
+
+	reqs := make([]*kv.Request, 0, (len(c.TableInfo.Indices)+1)*(len(partDefs)+1))
+	if err := c.appendRequest(ctx, c.TableInfo.ID, &reqs); err != nil {
```

Comment:
Can we save the request to the table when it is a partitioned table since it won't have any data?

Reply:
Yes we could (add an `if` clause), but since the range contains no data I think the overhead would be small. OTOH if we do want to support global index, I suppose those indices will be placed in the table itself, so we still need to `appendRequest(c.TableInfo.ID)`.